crates/circuit/src/circuit_data.rs

```diff
@@ -2751,6 +2840,10 @@ impl CircuitData {
         .iter()
         .map(|stretch| self.stretches.get(*stretch).unwrap())
     }
+    pub fn reverse(mut self) -> Self {
+        self.data.reverse();
+        self
+    }
 }
```

Comment:
If we're adding a method for this we need to document it, especially since it's an inplace modification.

That being said this method isn't safe, if nothing else it would result in an invalid parameter table because the instruction indices are no longer correct. Realistically this isn't a safe mutation to do in place because the indices are significant and are shared references elsewhere in the circuit. If we need a method to reverse this and just a reverse iterator (which we can do with something like CircuitData::data().iter().rev()`) isn't sufficient we probably need to do:

```suggestion
    pub fn reverse(&self) -> Self {
        let mut out = self.clone_empty_like();
        for inst in self.data().iter().rev() {
            out.push(inst.clone())
        }
        out
    }
}
```

Reply:
Used your suggestion (the syntax is slightly different)